pkr/driver/docker.py

```diff
@@ -63,70 +65,157 @@ def get_meta(self, extras, kard):
-                }
-            )
+            # Legacy format
+            if build_cfg is run_cfg is None:
+                context = container.get("context", self.DOCKER_CONTEXT)
+
+                # Process requirements
+                for src in self.kard.env.get_requires([container_name]):
+                    template = {
+                        "source": src["origin"],
+                        "origin": src["origin"],
+                        "destination": src["dst"],
+                        "subfolder": context,
+                        "excluded_paths": src.get("exclude", []),
+                        "gen_template": False,
+                    }
+                    templates.add(HashableDict(template))
+
+                try:
+                    dockerfile = container["dockerfile"]
+                except KeyError:
+                    # In this case, we use an image provided by the hub
+                    continue
 
-            # Also add anything matching dockerfile name stem with an extension
-            # suffix of some kind
-            templates.append(
-                {
-                    "source": templates_path / f"{dockerfile}.*",  # Match template
-                    "origin": templates_path,
-                    "destination": "",
-                    "subfolder": context,
-                }
-            )
+                # Automatically add dockerfile name matching folder to the context
+                dockerfile = Path(dockerfile).stem
+                templates.add(
+                    HashableDict(
+                        {
+                            "source": str(templates_path / f"{dockerfile}"),  # Match template
+                            "origin": str(templates_path),
+                            "destination": "",
+                            "subfolder": context,
+                        }
+                    )
+                )
+
+                # Also add anything matching dockerfile name stem with an extension
+                # suffix of some kind
+                templates.add(
+                    HashableDict(
+                        {
+                            "source": str(templates_path / f"{dockerfile}.*"),  # Match template
+                            "origin": str(templates_path),
+                            "destination": "",
+                            "subfolder": context,
+                        }
+                    )
+                )
 
-        return templates
+        return list(templates)
 
     def context_path(self, sub_path, container):
         """Return absolute path for sub_path relative to container context"""
-        context = self.kard.env.get_container(container).get("context", self.DOCKER_CONTEXT)
+        if not (
+            context := self.kard.env.get_container(container).get("build", {}).get("subfolder")
+        ):
+            context = self.kard.env.get_container(container).get("context", self.DOCKER_CONTEXT)
```

Comment:
Here and in the next blocks, the walrus'ed variable is not used inside the conditional, so this seems less readable to me than 
```
cnt = self.kard.env.get_container(container)
context = cnt.get("build", {}).get("subfolder") or cnt.get("context", self.DOCKER_CONTEXT)
```